api/v1beta1/tags.go

```diff
@@ -0,0 +1,157 @@
+)
+
+// ResourceManagerTags is an slice of ResourceManagerTag structs.
+type ResourceManagerTags []ResourceManagerTag
+
+// ResourceManagerTagsMap defines a map of key value pairs as expected by compute.InstanceParams.ResourceManagerTags.
+type ResourceManagerTagsMap map[string]string
+
+// ResourceManagerTag is a tag to apply to GCP resources managed by the GCP provider.
+type ResourceManagerTag struct {
+	// ParentID is the ID of the hierarchical resource where the tags are defined
+	// e.g. at the Organization or the Project level. To find the Organization or Project ID ref
+	// https://cloud.google.com/resource-manager/docs/creating-managing-organization#retrieving_your_organization_id
+	// https://cloud.google.com/resource-manager/docs/creating-managing-projects#identifying_projects
+	// An OrganizationID must consist of decimal numbers, and cannot have leading zeroes.
+	// A ProjectID must be 6 to 30 characters in length, can only contain lowercase letters,
+	// numbers, and hyphens, and must start with a letter, and cannot end with a hyphen.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=32
+	// +kubebuilder:validation:Pattern=`(^[1-9][0-9]{0,31}$)|(^[a-z][a-z0-9-]{4,28}[a-z0-9]$)`
+	ParentID string `json:"parentID"`
+
+	// Key is the key part of the tag. A tag key can have a maximum of 63 characters and cannot
+	// be empty. Tag key must begin and end with an alphanumeric character, and must contain
+	// only uppercase, lowercase alphanumeric characters, and the following special
+	// characters `._-`.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=`^[a-zA-Z0-9]([0-9A-Za-z_.-]{0,61}[a-zA-Z0-9])?$`
+	Key string `json:"key"`
+
+	// Value is the value part of the tag. A tag value can have a maximum of 63 characters and
+	// cannot be empty. Tag value must begin and end with an alphanumeric character, and must
+	// contain only uppercase, lowercase alphanumeric characters, and the following special
+	// characters `_-.@%=+:,*#&(){}[]` and spaces.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=`^[a-zA-Z0-9]([0-9A-Za-z_.@%=+:,*#&()\[\]{}\-\s]{0,61}[a-zA-Z0-9])?$`
+	Value string `json:"value"`
+}
+
+// Merge merges resource manager tags in receiver and other.
+func (t *ResourceManagerTags) Merge(other ResourceManagerTags) {
+	*t = append(*t, other...)
+}
+
+// Bind creates a TagBinding between a TagValue and a Google Cloud resource.
+// If any of the SDK calls fail, the error is logged and no action is taken.
+func (t ResourceManagerTags) Bind(ctx context.Context, location, name string) {
+	log := log.FromContext(ctx)
+	endpoint := fmt.Sprintf("%s-cloudresourcemanager.googleapis.com:443", location)
+	c, err := resourcemanager.NewTagBindingsClient(
```

Comment:
When creating the client for the resource manager we should do something similar to how we create other clients:

https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/8bb4d822317484d2f0e1cafc4fee6bf17376ed80/cloud/scope/clients.go#L91

This will ensure that "per cluster" credentials are handled.